vars/npmExecute.groovy

```diff
@@ -0,0 +1,49 @@
+import static com.sap.piper.Prerequisites.checkScript
+import com.sap.piper.ConfigurationHelper
+import com.sap.piper.Utils
+import groovy.transform.Field
+
+@Field def STEP_NAME = getClass().getName()
+@Field Set GENERAL_CONFIG_KEYS = []
+@Field Set STEP_CONFIG_KEYS = ['dockerImage', 'defaultNpmRegistry', 'npmScript']
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS + ['dockerOptions']
+
+def call(Map parameters = [:], body) {
+    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters) {
+
+        final script = checkScript(this, parameters) ?: this
+
+        // load default & individual configuration
+        Map configuration = ConfigurationHelper.newInstance(this)
+            .loadStepDefaults()
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS)
+            .mixinStageConfig(script.commonPipelineEnvironment, parameters.stageName?:env.STAGE_NAME, STEP_CONFIG_KEYS)
+            .mixin(parameters, PARAMETER_KEYS)
+            .use()
+
+        new Utils().pushToSWA([
+            step: STEP_NAME,
+            stepParamKey1: 'scriptMissing',
+            stepParam1: parameters?.script == null
+        ], configuration)
+
+        try {
+            if (configuration.defaultNpmRegistry) {
+                sh "npm config set registry ${configuration.defaultNpmRegistry}"
```

Comment:
This should be part of the project in an `.npmrc` file

Reply:
agree, deleted.